packages/dev/gui/src/2D/advancedDynamicTexture.ts

```diff
@@ -293,30 +293,30 @@ export class AdvancedDynamicTexture extends DynamicTexture {
      * @param predicate defines an optional predicate that will be called on every evaluated child, the predicate must return true for a given child to be part of the result, otherwise it will be ignored
      * @returns all child controls
      */
-    public getDescendants(directDescendantsOnly?: boolean, predicate?: (control: Control) => boolean): Control[] {
-        return this._rootContainer.getDescendants(directDescendantsOnly, predicate);
+    public getDescendants<T extends Control>(directDescendantsOnly?: boolean, predicate?: (control: Control) => boolean): T[] {

```

Comment:
I'm wondering if it makes sense to have the single type name here, as it is not guaranteed that all descendants of a control will have the same type...